batch/src/main/java/org/openmrs/analytics/JdbcFetchUtil.java

```diff
@@ -0,0 +1,205 @@
+			this.resourceType = resourceType;
+		}
+		
+		@Override
+		public PCollection<SearchSegmentDescriptor> expand(PCollection<String> uuIds) {
+			return uuIds
+			        // create KV required by GroupIntoBatches
+			        .apply(ParDo.of(new DoFn<String, KV<String, String>>() {
+				        
+				        @ProcessElement
+				        public void processElement(ProcessContext c) {
+					        if (c.element() != null) {
+						        c.output(KV.of(resourceType, c.element()));
+					        }
+				        }
+			        }))
+			        .apply(String.format("GroupIntoBatches of %s", this.batchSize),
+			            GroupIntoBatches.<String, String> ofSize(this.batchSize))
+			        .apply("Generate Segments", ParDo.of(new DoFn<KV<String, Iterable<String>>, SearchSegmentDescriptor>() {
+				        
+				        @ProcessElement
+				        public void processElement(@Element KV<String, Iterable<String>> element,
+				                OutputReceiver<SearchSegmentDescriptor> r) {
+					        List<String> uuIds = new ArrayList<String>();
+					        element.getValue().forEach(uuIds::add);
+					        r.output(SearchSegmentDescriptor.create(
+					            String.format("%s?_id=%s", baseBundleUrl, String.join(",", uuIds)), uuIds.size()));
+					        
+				        }
+			        }));
+		}
+	}
+	
+	public ResultSet FetchMaxId(String tableName) throws SQLException {
+		String tableId = tableName + "_id";
+		Connection con = this.comboPooledDataSource.getConnection();
+		Statement statement = con.createStatement();
+		ResultSet resultSet = statement.executeQuery(String.format("SELECT MAX(`%s`) from %s", tableId, tableName));
+		return resultSet;
+	}
+	
+	public JdbcIO.DataSourceConfiguration getJdbcConfig() {
+		return JdbcIO.DataSourceConfiguration.create(this.comboPooledDataSource);
+	}
+	
+	public Map<String, Integer> CreateIdRanges(int count, int rangeSize) {
+		int ranges = count / rangeSize;
+		Map<String, Integer> rangeMap = new HashMap<String, Integer>();
+		for (int i = 0; i < ranges; i++) {
+			int rangeFrom = i * rangeSize;
+			int rangeTo = (i + 1) * rangeSize;
+			String range = String.format("%s,%s", rangeFrom, rangeTo);
+			rangeMap.put(range, rangeFrom);
+		}
+		if (count > ranges * rangeSize) {
+			int rangeFrom = ranges * rangeSize;
+			int rangeTo = ranges * rangeSize + count % rangeSize;
+			String range = String.format("%s,%s", rangeFrom, rangeTo);
+			rangeMap.put(range, rangeFrom);
+		}
+		return rangeMap;
+	}
+	
+	public Map<String, String> createFhirReverseMap(String searchString, String tableFhirMapPath) throws IOException {
+		Gson gson = new Gson();
+		Path pathToFile = Paths.get(tableFhirMapPath);
+		try (Reader reader = Files.newBufferedReader(pathToFile.toAbsolutePath(), StandardCharsets.UTF_8)) {
+			GeneralConfiguration generalConfiguration = gson.fromJson(reader, GeneralConfiguration.class);
+			LinkedHashMap<String, EventConfiguration> tableToFhirMap = generalConfiguration.getEventConfigurations();
```

Comment:
Do we need a `LinkedHashMap` or is a `HashMap` enough?

Reply:
we do not, I have switched to HashMap, thanks!